python-avd/pyavd/_eos_designs/structured_config/network_services/vlan_interfaces.py

```diff
@@ -149,6 +149,14 @@ def _get_vlan_interface_config_for_svi(
             # If any anycast IPs are set, we also enable link-local IPv6 per best practice, unless specifically disabled with 'ipv6_enable: false'
             vlan_interface_config.ipv6_enable = default(vlan_interface_config.ipv6_enable, True)  # noqa: FBT003
 
+            if svi.ipv6_nd.advertise_ipv6_address_virtuals:
+                nd_prefix_options: dict[str, Any] = {
+                    "valid_lifetime": svi.ipv6_nd.valid_lifetime,
+                    "preferred_lifetime": svi.ipv6_nd.preferred_lifetime,
+                }
+                for ipv6_address in svi.ipv6_address_virtuals:
+                    vlan_interface_config.ipv6_nd_prefixes.append_new(ipv6_prefix=ipv6_address, **nd_prefix_options)
```

Comment:
We loose all type safety if you go in and out of a dict with vague type hints (Any). Please change to this, which is much more readable and probably also more efficient.
```suggestion
            for ipv6_address in svi.ipv6_address_virtuals:
                vlan_interface_config.ipv6_nd_prefixes.append_new(
                    ipv6_prefix=ipv6_address,
                    valid_lifetime=svi.ipv6_nd.valid_lifetime,
                    preferred_lifetime=svi.ipv6_nd.preferred_lifetime,
```